internal/stremio/store/stream.go

```diff
@@ -62,10 +64,30 @@ func handleStream(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	ctx := r.Context()
+	ctx = context.WithValue(ctx, "user_data", ud)
+
 	videoIdWithLink := getId(r)
 	contentType := r.PathValue("contentType")
 	isStremThruStoreId := isStoreId(videoIdWithLink)
 	isImdbId := strings.HasPrefix(videoIdWithLink, "tt")
+
+	// Add logging to debug the request details
+	log.Info("Stream request details",
+		"path", r.URL.Path,
+		"catalogs_only", ud.CatalogsOnly,
+		"content_type", contentType,
+		"video_id", videoIdWithLink)
+
+	// Block streams if CatalogsOnly is enabled and the request is not from store
+	if ud.CatalogsOnly && !isStremThruStoreId {
```

Comment:
Also, the manifest should be modified to not accept `tt` id prefix. Otherwise Stremio will keep sending streams request to the addon.